File: FrogJmp/obj-c/FrogJmp/FrogJmp.m
//
// FrogJmp.m
// Codility - FrogJmp
//

#import "FrogJmp.h"

@implementation FrogJmp

int solution(int X, int Y, int D) {
    
    int result = 0;
    int distance = Y - X;
    int rest = 0;
    
    result = distance / D;
    rest = distance % D;
    
    if (rest != 0) {
        result++;
    }
    
    return result;
    
}

@end

File: FrogJmp/php/FrogJmp.php
<?php

namespace Codility;

class FrogJmp {

	function solution($X, $Y, $D) {

		$result = 0;
		$distance = $Y - $X;
		$rest = 0;

		$result = (int)($distance / $D);
		$rest = $distance % $D;

		if ($rest!=0) {
			$result++;
		}

		return $result;

	}
	
}